src/test/java/abbyy/cloudsdk/v2/client/ClientTest.java

```diff
@@ -17,450 +16,349 @@
-            TaskInfo businessCardProcessingTaskInfo2 = taskDeletionTask.get();
-            Assert.assertEquals(0, taskInfoComparator.compare(businessCardProcessingTaskInfo, businessCardProcessingTaskInfo2));
-            Assert.assertEquals(TaskStatus.Deleted, businessCardProcessingTaskInfo2.getStatus());
-
-            // checking finished task list
-            CompletableFuture<TaskList> finishedTaskListTask = ocrClient.listFinishedTaskAsync();
-            TaskList finishedTaskList = finishedTaskListTask.get();
-            if (finishedTaskList.getTasks() != null) {
-                contains = finishedTaskList.getTasks().stream().anyMatch(taskInfo -> taskInfoComparator.compare(taskInfo, businessCardProcessingTaskInfo) == 0);
-                Assert.assertFalse(contains);
-            }
         } catch (FileNotFoundException e) {
             Assume.assumeNoException(e);
         }
     }
 
     @Test
-    public void processTextFieldTest() throws InterruptedException, ExecutionException {
+    public void processBarcodeFieldTest()  throws InterruptedException, ExecutionException {
         try {
-            File file = new File(TEXTFIELD_FILEPATH);
-            FileInputStream fileStream = new FileInputStream(file);
+            FileInputStream fileStream = new FileInputStream(TestFile.FIELDS);
 
-            // creating text field processing task
-            TextFieldProcessingParams textFieldProcessingParams = new TextFieldProcessingParams();
-            textFieldProcessingParams.setDescription("Test textfield");
+            BarcodeFieldProcessingParams barcodeFieldProcessingParams = new BarcodeFieldProcessingParams();
+            barcodeFieldProcessingParams.setBarcodeTypes(new BarcodeType[]{BarcodeType.Ean8});
+            barcodeFieldProcessingParams.setRegion("1800,3200,2250,3400");
 
-            CompletableFuture<TaskInfo> textFieldProcessingTask = ocrClient.processTextFieldAsync(textFieldProcessingParams, fileStream, file.getName(), true);
-            TaskInfo textFieldProcessingTaskInfo = textFieldProcessingTask.get();
+            TaskInfo processBarcodeFieldTask = apiClient.processBarcodeFieldAsync(
+                    barcodeFieldProcessingParams,
+                    fileStream,
+                    TestFile.FIELDS,
+                    true
+            ).get();
 
-            Assert.assertNotNull(textFieldProcessingTaskInfo.getTaskId());
-            Assert.assertEquals(TaskStatus.Completed, textFieldProcessingTaskInfo.getStatus());
-            Assert.assertEquals(1, textFieldProcessingTaskInfo.getResultUrls().size());
+            checkResultTask(processBarcodeFieldTask);
 
-            // checking task status
-            CompletableFuture<TaskInfo> taskStatusTask = ocrClient.getTaskStatusAsync(textFieldProcessingTaskInfo.getTaskId());
-            Assert.assertEquals(TaskStatus.Completed, taskStatusTask.get().getStatus());
+        } catch (FileNotFoundException e) {
```

Comment:
Why do we need these checks in all tests?

Reply:
I use this approach to skip tests, concerning external files, if the files were not found. In this case, test working with missing files are not marked as FAILED.
Of course, I can just declare FileNotFoundException as throwable by the test method, if you want to see it like this.

Reply:
In my opinion, it isn't a healthy situation for tests so let's let them be failed